test/test_core.py

```diff
@@ -3141,20 +3141,24 @@ def test_nestedstructs(self):
       expected = '*16,0,4,8,8,12|20,0,4,4,8,12,12,16|24,0,20,0,4,4,8,12,12,16*\n*0,0,1,2,64,68,69,72*\n*2*'
     self.do_run(src, expected)
 
-  def prep_dlfcn_main(self):
+  def prep_dlfcn_main(self, libs=None):
+    if libs is None:
+      libs = ['liblib.so']
     self.clear_setting('SIDE_MODULE')
     # Link against the side modules but don't load them on startup.
     self.set_setting('NO_AUTOLOAD_DYLIBS')
-    self.emcc_args.append('liblib.so')
+    self.emcc_args += libs
     # This means we can use MAIN_MODULE=2 without needing to explicitly
     # specify EXPORTED_FUNCTIONS.
     self.set_setting('MAIN_MODULE', 2)
 
-  def build_dlfcn_lib(self, filename):
+  def build_dlfcn_lib(self, filename, outfile='liblib.so', emcc_args=None):
     self.clear_setting('MAIN_MODULE')
     self.set_setting('SIDE_MODULE')
-    outfile = self.build(filename, js_outfile=not self.is_wasm())
-    shutil.move(outfile, 'liblib.so')
+    cmd = [compiler_for(filename), filename, '-o', outfile] + self.get_emcc_args()
+    if emcc_args:
+      cmd += emcc_args
```

Comment:
Why do we need this in addition to `get_emcc_args()` on the prevoius line?

Reply:
I used it below on the line `self.build_dlfcn_lib('liba.c', outfile='liba.so', emcc_args=['libb.so'])`

Here i want to set `libb.so` as an argument to compile step, but I don't want to add it to all compile steps that make up the test.

We have several functions that both take emcc_args and honor the global emcc_args.   Sometimes it makes more sense to use one over the other I think.

Reply:
I see, thanks, I was just confused by the name then.